### PR TITLE
core/sstring: do not use ostream_formatter

### PR DESCRIPTION
**Pull request by tchaikov**
use fmt::formatter<std::string_view> instead, as fmt::ostream_formatter creates an extra basic_memory_buffer and a std::basic_streambuf on top of it, both of this incurs extra overhead. but sstring is used widely in a Seastar application. so, in this change, let's just reuse fmt::formatter<std::string_view> for formatting sstring for better performance.

### CODE REVIEW

File: include/seastar/core/sstring.hh [hunk 1]
```diff
@@ -781,9 +781,14 @@ std::ostream& operator<<(std::ostream& os, const std::unordered_map<Key, T, Hash
 }
 }
 
-#if FMT_VERSION >= 90000
+#if FMT_VERSION >= 90000 && !defined(FMT_DEPRECATED_OSTREAM)
```
Comment (avikivity) [inline include/seastar/core/sstring.hh:784]: Why is this needed? Doesn't the formatter take precedence when FMT_DEPRECATED_OSTREAM is defined?

Comment (tchaikov) [reply include/seastar/core/sstring.hh]: > Why is this needed? 

no, it's not needed. i was misled by an error which was not relevant. 

> Doesn't the formatter take precedence when FMT_DEPRECATED_OSTREAM is defined?

agreed. reverted this change.

**Comment by tchaikov** [comment]
changelog

- dropped the check for `FMT_DEPRECATED_OSTREAM`

@avikivity hi Avi, could you help re-review this change?
